ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/show/status/formatter/ShowTableStatusFormatter.java

```diff
@@ -89,7 +93,18 @@ List<Path> getLocations(Hive db, Partition partition, Table table) throws HiveEx
     List<Path> locations = new ArrayList<Path>();
     if (table.isPartitioned()) {
       if (partition == null) {
-        for (Partition currPartition : db.getPartitions(table)) {
+      List<Partition> partitions;
+      GetProjectionsSpec getProjectionsSpec = new GetPartitionProjectionsSpecBuilder()
+              .addProjectFieldList(Arrays.asList("sd.location")).build();
+        GetPartitionsRequest request = new GetPartitionsRequest(table.getDbName(), table.getTableName(), getProjectionsSpec, null);
```

Comment:
can we use `GetPartitionsRequestBuilder` here, maybe add simpler constructor

Reply:
I am a bit skeptical about adding the new constructor for this use-case, as the code complexity approximately would be the same here. GetPartitionsRequestBuilder was built so that we would not have to write multiple set_methods for our projections,